src/main/kotlin/org/jitsi/jicofo/xmpp/JibriIqHandler.kt

```diff
@@ -0,0 +1,55 @@
+/*
+ * Jicofo, the Jitsi Conference Focus.
+ *
+ * Copyright @ 2017-Present 8x8, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jitsi.jicofo.xmpp
+
+import org.jitsi.jicofo.recording.jibri.CommonJibriStuff
+import org.jitsi.jicofo.util.ErrorResponse
+import org.jitsi.xmpp.extensions.jibri.JibriIq
+import org.jivesoftware.smack.iqrequest.AbstractIqRequestHandler
+import org.jivesoftware.smack.iqrequest.IQRequestHandler
+import org.jivesoftware.smack.packet.IQ
+import org.jivesoftware.smack.packet.XMPPError
+import java.util.*
+
+/**
+ * A Smack [IQRequestHandler] for "jibri" IQs. Terminates all "jibri" IQs received by Smack, but delegates their
+ * handling to specific [CommonJibriStuff] instances.
+ */
+class JibriIqHandler()
+    : AbstractIqRequestHandler(JibriIq.ELEMENT_NAME, JibriIq.NAMESPACE, IQ.Type.set, IQRequestHandler.Mode.async) {
+
+    private val jibris = Collections.synchronizedList(LinkedList<CommonJibriStuff>())
+
+    fun addJibri(jibri: CommonJibriStuff) {
+        jibris.add(jibri)
+    }
+
+    fun removeJibri(jibri: CommonJibriStuff?) {
+        jibris.remove(jibri)
+    }
+
+    /**
+     * {@inheritDoc}
+     * Pass the request to the first [CommonJibriStuff] that wants it.
+     */
+    override fun handleIQRequest(iq: IQ): IQ = synchronized(jibris) {
```

Comment:
Is this `synchronized(jibris)` necessary here? It's not used in `addJibri` or `removeJibri`.

Reply:
Yes, the `Collections.synchronizedList` docs specifically mention that it is required for iteration.

Reply:
Wow, interesting...so it must do its own protection on that same lock inside of add and remove or something?  I'd say it's worth a comment here.  Or what about using `CopyOnWriteArrayList` instead?  It's weird to me that _only_ iteration requires synchronizing, I'd think it would be easy for someone to get tripped up when reading this.

Reply:
I just made that change, but I realized that each conference adds a handler here, even if jibri is not used, so under stress there will be a lot of copying.

Reply:
The copy would just be on `addJibri`/`removeJibri`, which isn't related to the amount of handlers?  Or maybe I'm missing something.

Reply:
The list is comparable in size to the number of conferences, and it will be copied any time a conferences is started/stoppe

Reply:
Ok.  conference start/stop doesn't seem to me like a common enough operation that this would be a problem, but I won't insist on the change.  A comment then?  I just imagine it will be very tempting to think there's a synchronization issue here to someone who's reading it.